packages/amplify-category-hosting/__tests__/lib/S3AndCloudFront/helpers/cloudfront-manager.test.js

```diff
@@ -0,0 +1,75 @@
+
+describe('cloudfront-manager', () => {
+    const mockAmplifyMeta = {
+        "providers": {
+            "awscloudformation": {
+                "AuthRoleName": "checkhosting-20190226163640-authRole",
+                "UnauthRoleArn": "arn:aws:iam::mockAccountId:role/checkhosting-20190226163640-unauthRole",
+                "AuthRoleArn": "arn:aws:iam::mockAccountId:role/checkhosting-20190226163640-authRole",
+                "Region": "us-west-2",
+                "DeploymentBucketName": "checkhosting-20190226163640-deployment",
+                "UnauthRoleName": "checkhosting-20190226163640-unauthRole",
+                "StackName": "checkhosting-20190226163640",
+                "StackId": "arn:aws:cloudformation:us-west-2:mockAccountId:stack/checkhosting-20190226163640/2c061610-3a28-11e9-acf3-02ee71065ed8"
+            }
+        },
+        "hosting": {
+            "S3AndCloudFront": {
+                "service": "S3AndCloudFront",
+                "providerPlugin": "awscloudformation",
+                "providerMetadata": {
+                    "s3TemplateURL": "https://s3.amazonaws.com/checkhosting-20190226163640-deployment/amplify-cfn-templates/hosting/template.json",
+                    "logicalId": "hostingS3AndCloudFront"
+                },
+                "lastPushTimeStamp": "2019-02-27T00:39:17.966Z",
+                "output": {
+                    "S3BucketSecureURL": "https://checkosting-20190226163802-hostingbucket-dev.s3.amazonaws.com",
+                    "WebsiteURL": "http://checkosting-20190226163802-hostingbucket-dev.s3-website-us-west-2.amazonaws.com",
+                    "Region": "us-west-2",
+                    "HostingBucketName": "checkosting-20190226163802-hostingbucket-dev", 
+                    "CloudFrontDistributionID": "mockCloudFrontDistributionID",
+                    "CloudFrontSecureURL": "mockCloudFrontSecureURL"
+
+                },
+                "lastPushDirHash": "83Bhmmec48dILMj3mi2T25B4700="
+            }
+        }
+    }
+
+    const mockContext = {
+        amplify: {
+            getProviderPlugins: jest.fn(()=>{
+                return {
+                    "awscloudformation": "mockAwsProviderModule"
+                };
+            })
+        },
+        print: {
+            info: jest.fn(),
+            warning: jest.fn(),
+            error: jest.fn(),
+            success: jest.fn()
+        },
+        parameters: {
+            options: {
+                invalidateCache: true
+            }
+        },
+        exeInfo: {
+            serviceMeta: mockAmplifyMeta.hosting.S3AndCloudFront
+        }
+    }; 
+
+    beforeAll(() => {
+    }); 
+
+    beforeEach(() => {
+    });
+
+    test('invalidateCloudFront', async () => {
+        const result = await cloudFrontManager.invalidateCloudFront(mockContext); 
+        expect(result).toBe(mockContext); 
+        expect(mockContext.exeInfo.cftInvalidationData).toBeDefined(); 
```

Comment:
Don't you want to test `mockAwsProviderModule.createInvalidation` is called?

Reply:
I omitted it because it take a little more extra effort to setup, but it was check by the line 
expect(mockContext.exeInfo.cftInvalidationData).toBeDefined(); 

I added the function call check in the follow up dates.